### ISSUE TITLE
Prototype sending data via VS OTel collector

### ISSUE THREAD

**Issue by JanKrivanek** [dotnet/msbuild#10945]
### Motivation

We want to evaluate viability and benefits of using the new VS OTel collector.
Specifically this should help us sample and filter-out/in data that would otherwise be of a too high volume for us.

### Goals
 * Prototype sending data from MSBuild (idealy those defined in https://github.com/dotnet/msbuild/issues/10946) via VS OTel collector (doesn't need to be mergable, can be runnable just localy with manual setup)
 * Experiment with and document the head filtering capability

### Details
* https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/43901/OpenTelemetry-in-VS
* Packages:
  * https://devdiv.visualstudio.com/DevDiv/_artifacts/feed/VS/NuGet/Microsoft.VisualStudio.OpenTelemetry.ClientExtensions/overview/0.1.700-beta
  *  https://devdiv.visualstudio.com/DevDiv/_artifacts/feed/VS/NuGet/Microsoft.VisualStudio.OpenTelemetry.Collector/overview/0.1.700-beta

**Comment by JanKrivanek** [comment]
Some playing around: https://github.com/JanKrivanek/otel-test-01


![Image](https://github.com/user-attachments/assets/6c6c5701-0c5e-424d-97ba-9f731c234999)

**Comment by JanProvaznik** [comment]
Documented learnings in https://github.com/dotnet/msbuild/pull/11175 and there are data being sent to the backend in the implementation https://github.com/dotnet/msbuild/pull/11255/ when opted in.

Tl;dr for now sampling has to be done outside of the VS OTel libraries, to prevent new assembly loads (performance).
As of now there is not a mechanism for filtering.
